regxmllib/CMakeLists.txt

```diff
@@ -1,29 +1,92 @@
-cmake_minimum_required (VERSION 3.5)
+cmake_minimum_required (VERSION 3.6)
 project (regxmllibc)
+option(BUILD_SHARED_LIB Bool OFF)
+
+# Set build type
+if(BUILD_SHARED_LIB)
+set(LIB_TYPE SHARED)
+if(WIN32 AND NOT CYGWIN)
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
```

Comment:
Can't this be set for all platforms... or does it cause problems in cygwin?

Reply:
It's being set specifically for WIN32 DLLs. I did not test with cygwin and have it therefore explicitly excluded.